compiler/test/dotty/tools/dotc/ParallelTesting.scala

```diff
@@ -406,56 +406,63 @@ trait ParallelTesting {
 
   private final class RunTest(testSources: List[TestSource], times: Int, threadLimit: Option[Int], suppressAllOutput: Boolean)
   extends Test(testSources, times, threadLimit, suppressAllOutput) {
-    private def verifyOutput(checkFile: JFile, dir: JFile, testSource: TestSource, warnings: Int) = try {
+
+    private def runMain(dir: JFile, testSource: TestSource): Array[String] = {
+      import java.io.ByteArrayOutputStream
+      import java.net.{ URL, URLClassLoader }
+
+      val printStream = new ByteArrayOutputStream
+      try {
         // Do classloading magic and running here:
-        import java.net.{ URL, URLClassLoader }
-        import java.io.ByteArrayOutputStream
         val ucl = new URLClassLoader(Array(dir.toURI.toURL))
         val cls = ucl.loadClass("Test")
         val meth = cls.getMethod("main", classOf[Array[String]])
 
-        val printStream = new ByteArrayOutputStream
         Console.withOut(printStream) {
           meth.invoke(null, Array("jvm")) // partest passes at least "jvm" as an arg
         }
+      }
+      catch {
+        case ex: NoSuchMethodException =>
+          echo(s"test in '$dir' did not contain method: ${ex.getMessage}      ")
```

Comment:
It would be also nice to keep the stack trace. 
In case method was expected to be a forwarder it's good to know from where is it called.
Similar in other cases.